File: Wavelength/SWMessagesLayout.m
//
//  SWMessagesLayout.m
//  hashtag
//
//  Created by Ethan Sherr on 10/10/14.
//  Copyright (c) 2014 Ethan Sherr. All rights reserved.
//

#import "SWMessagesLayout.h"

@implementation SWMessagesLayout

-(NSArray*)layoutAttributesForElementsInRect:(CGRect)rect
{
    NSMutableArray *attributesArray = [[super layoutAttributesForElementsInRect:rect] mutableCopy];
    
//    NSMutableIndexSet *omittedSections = [NSMutableIndexSet indexSet];
//    NSLog(@"*attributesArya*");
    for (UICollectionViewLayoutAttributes *attributes in attributesArray)
    {
        //0 cell, 1 supplement, 2 decoration
//        NSLog(@"attributes.representedelementCatagory = %d [%d, %d]", attributes.representedElementCategory, attributes.indexPath.section, attributes.indexPath.row);

        if (attributes.representedElementCategory == UICollectionElementCategoryCell)
        {
//            [omittedSections addIndex:attributes.indexPath.section];
            attributes.zIndex = -1;
//            attributes.alpha = 0.1f;
        } else
        {
            attributes.zIndex = 1;
//            attributes.alpha = 0.1;
        }
    }
//    for (UICollectionViewLayoutAttributes *attributes in attributesArray) {
//        if ([attributes.representedElementKind isEqualToString:UICollectionElementKindSectionFooter]) {
//            [omittedSections removeIndex:attributes.indexPath.section];
//        }
//    }
//    [omittedSections enumerateIndexesUsingBlock:^(NSUInteger idx, BOOL *stop) {
//        NSIndexPath *indexPath = [NSIndexPath indexPathForItem:0 inSection:idx];
//        UICollectionViewLayoutAttributes *attributes = [self layoutAttributesForSupplementaryViewOfKind:UICollectionElementKindSectionFooter
//                                                                                            atIndexPath:indexPath];
//        [attributesArray addObject:attributes];
//    }];
//    for (UICollectionViewLayoutAttributes *attributes in attributesArray) {
//        if ([attributes.representedElementKind isEqualToString:UICollectionElementKindSectionFooter]) {
//            
//            // adjust any aspect of each header's attributes here, including frame or zIndex
//            attributes.zIndex = 100;
//            
//        }
//        
//    }
    
    return attributesArray;
}

//-(UICollectionViewLayoutAttributes*)layoutAttributesForDecorationViewOfKind:(NSString *)elementKind atIndexPath:(NSIndexPath *)indexPath
//{
//    UICollectionViewLayoutAttributes *attributes = [super layoutAttributesForDecorationViewOfKind:elementKind atIndexPath:indexPath];
//    attributes.zIndex = 100;
//    return attributes;
//}
//-(UICollectionViewLayoutAttributes*)layoutAttributesForSupplementaryViewOfKind:(NSString *)elementKind atIndexPath:(NSIndexPath *)indexPath
//{
//    UICollectionViewLayoutAttributes *attributes = [super layoutAttributesForSupplementaryViewOfKind:elementKind atIndexPath:indexPath];
//    attributes.zIndex = 100000;
//    return attributes;
//}
//-(UICollectionViewLayoutAttributes*)layoutAttributesForItemAtIndexPath:(NSIndexPath *)indexPath
//{
//    UICollectionViewLayoutAttributes *att = [super layoutAttributesForItemAtIndexPath:indexPath];
//    att.zIndex = 0;
//    return att;
//}

- (BOOL) shouldInvalidateLayoutForBoundsChange:(CGRect)newBound {
    
    return YES;
    
}







@end

File: Shortwave/SWMessagesLayout.swift
//
//  SWMessagesLayout.swift
//  Shortwave
//
//  Created by Ethan Sherr on 7/27/14.
//  Copyright (c) 2014 Ethan Sherr. All rights reserved.
//

import UIKit

class SWMessagesLayout: UICollectionViewFlowLayout {

    
    override func layoutAttributesForItemAtIndexPath(indexPath: NSIndexPath!) -> UICollectionViewLayoutAttributes!
    {
        let attribute = super.layoutAttributesForItemAtIndexPath(indexPath)

//        let size = self.collectionViewContentSize()
//        attribute.frame.origin.y = size.height - attribute.frame.origin.y - attribute.frame.size.height
        
        return attribute
    }
    
    override func collectionViewContentSize() -> CGSize
    {
        var size = super.collectionViewContentSize()
//        if size.height < collectionView.frame.size.height
//        {
//            size.height = collectionView.frame.size.height
//        }
        return size
        
    }
    
    
    
//    override func layoutAttributesForElementsInRect(rect: CGRect) -> [AnyObject]!
//    {
//        //shift the rectangle down!
//        let size = self.collectionViewContentSize()
//        
//        let rect = CGRectMake(rect.origin.x, size.height - rect.size.height, rect.size.width, rect.size.height);
//        
//        println("rect = \(rect)")
//        let attributes:[AnyObject] = super.layoutAttributesForElementsInRect(rect)
//        
//        
////        println("size \(size)")
//        for attribute in attributes as [UICollectionViewLayoutAttributes]
//        {
////            println("size = \(size)")
////            println("attribute.y = \(attribute.frame.origin.y)")
//            attribute.frame.origin.y = size.height - attribute.frame.origin.y //- attribute.frame.size.height
//            println("aftr.y = \(attribute.frame.origin.y)")
//            
//            
//        }
//
//        return attributes;
//        
//      
//        
//    }

}
